geode-core/src/main/java/org/apache/geode/internal/cache/RegionService.java

```diff
@@ -27,4 +27,11 @@
    * @param region The region being destroyed
    */
   public void beforeRegionDestroyed(Region region);
+
```

Comment:
You introduced this RegionService under internal package. But there's RegionService at org.apache.geode.cache package. This class is better to be a different name, or merge its functionality into other RegionService.

Reply:
Thanks. Yes, its confusing. I can rename it. Any ideas? How about RegionCacheService?